app/demo-geant-integration/HitRootIO.cc

```diff
@@ -170,13 +179,29 @@ void HitRootIO::Close()
  */
 void HitRootIO::Merge()
 {
-    auto nthreads = G4RunManager::GetRunManager()->GetNumberOfThreads();
+#if G4VERSION_NUMBER >= 1070
+    auto const nthreads = G4RunManager::GetRunManager()->GetNumberOfThreads();
+#else
+    auto const nthreads = std::invoke([]() {
+        auto const runman{
+            dynamic_cast<G4MTRunManager*>(G4RunManager::GetRunManager())};
+        if (runman != nullptr)
```

Comment:
For dynamic casts, I prefer:
```suggestion
        if (auto* runman = dynamic_cast<G4MTRunManager*>(G4RunManager::GetRunManager()))
```
(note also using `auto*` instead of `auto`)